### DISCUSSION TITLE
ResponseValidationError with body

### DISCUSSION THREAD

**Discussion by gistart** [fastapi/fastapi#8934] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * Or, I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

# Response Validation

Documentation suggests to implement pydantic's `ValidationError` handler to catch response validation errors. Problem is, there's no access to actual response, input of pydantic model, via this exception.

On the other hand, these errors may be much harder to spot, even to reproduce. Ideally, I'd want to log/sentry both request and failed response in this case, but as far as I know, there's no way I can do it via exception_handler mechanism currently. The only option -- manually validate and handle errors in every request handler, instead of:
```python
return raw_dict

@app.exception_handler(ResponseValidationError)
def on_invalid_resp(req: Request, exc: ResponseValidationError):
   # once per project
```

I need to:
```python
# copy&paste to each handler
try: 
      return MyResponseModel.parse_obj(raw_dict)
except ValidationError:
     ...
```

So maybe it'll make sense to create `ResponseValidationError` subclass of ValidationError, with `body` containing failed object, just like it is implemented for request validation?

**Comment by efir-it** [comment]
Столкнулся с такой же проблемой, очень не информативна ошибка. Подскажите пожалуйста как вы ее решили?

**Reply by mitch-seymour** [reply]
I was able to get info about the error by following @gistart 's example. e.g.

```python
@router.get("/", response_model=List[EventSchema])
def show_records(db: Session = Depends(get_db)):
    records = db.query(Event).limit(5).all()
    try: 
        EventSchema.model_validate(records[0].__dict__)
    except Exception as e:
        print(e)
    return records
```

Just adapt that to your endpoint. The exception will contain more detailed info. e.g.

```sh
1 validation error for Event
some_field
  Input should be a valid string [type=string_type, input_value={}, input_type=JsonObject]
```

I agree with @gistart though, it would be nice to have this appear in the `ResponseValidationError` body.
